schema.go

```diff
@@ -110,26 +111,36 @@ func schemaOf(model reflect.Type) *Schema {
 	if model.Kind() != reflect.Struct {
 		panic("cannot construct parquet schema from value of type " + model.String())
 	}
-	schema = NewSchema(model.Name(), nodeOf(model, nil))
+	schema = NewSchema(model.Name(), nodeOf(model, StructTag{}, tags), SchemaOptions{
+		TagSource: tags,
+	})
 	if actual, loaded := cachedSchemas.LoadOrStore(model, schema); loaded {
 		schema = actual.(*Schema)
 	}
 	return schema
 }
 
+type SchemaOptions struct {
+	TagSource TagSource
+}
+
 // NewSchema constructs a new Schema object with the given name and root node.
 //
 // The function panics if Node contains more leaf columns than supported by the
 // package (see parquet.MaxColumnIndex).
-func NewSchema(name string, root Node) *Schema {
+func NewSchema(name string, root Node, options SchemaOptions) *Schema {
```

Comment:
I really think we should adopt the variadic argument options instead here, so the extra parameters are optional and we minimize the risk of breaking applications, for example:

```go
func NewSchema(name string, root Node, options ...SchemaOptions) *Schema {
  ...
}
```

Then, every call site doing `parquet.NewSchema` contains to remain syntactically correct.

I would also encourage us to pass the option list to `SchemaOf` so the new capability can also be used there.